### PR TITLE
Bugfix/video player

### PR DESCRIPTION
**Pull request by IvanSkadorva**

### CODE REVIEW

File: src/app/pages/video-player/progress-bar/progress-bar.component.scss [hunk 1]
```diff
@@ -20,30 +21,44 @@ $roboto-font-family: Roboto, 'Arial Narrow', Arial sans-serif;
       width: 79%;
     }
   }
+  &__progress-container:hover &__background {
+    height: 10px;
+    transition: all 0.4s ease 0s;
+  }
+  &__progress-container:hover &__play {
+    height: 10px;
+    transition: all 0.4s ease 0s;
```
Comment (Vladissqq) [inline src/app/pages/video-player/progress-bar/progress-bar.component.scss:30]: same code

File: src/app/pages/video-player/video-controllers/fullscreen-btn/fullscreen-btn.component.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<div class="fullscreen-btn" (click)="toggleFullscreen()">
```
Comment (Vladissqq) [inline src/app/pages/video-player/video-controllers/fullscreen-btn/fullscreen-btn.component.html:1]: why you use name container for div in choose-season.component , but here u use fulscreen-btn ?

File: src/app/pages/video-player/video-controllers/fullscreen-btn/fullscreen-btn.component.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        (isFullscreen: boolean): void => {
+          this.isFullscreen = isFullscreen;
+        },
+      ),
+    );
+  }
+
+  ngOnInit(): void {}
+
+  ngOnDestroy() {
```
Comment (Vladissqq) [inline src/app/pages/video-player/video-controllers/fullscreen-btn/fullscreen-btn.component.ts:26]: : void?

File: src/app/pages/video-player/video-controllers/video-controllers.component.scss [hunk 4]
```diff
@@ -1,4 +1,10 @@
 .controllers {
+  %button {
+    display: block;
+    background-size: contain;
+    padding: 5px;
+  }
+
   &__btns {
```
Comment (Vladissqq) [inline src/app/pages/video-player/video-controllers/video-controllers.component.scss:8]: i donnt like this class name. Can u change on more accurate class name?

File: src/app/pages/video-player/video-controllers/video-controllers.component.ts [hunk 5]
```diff
@@ -1,42 +1,38 @@
...
+  ngOnDestroy() {
+    this.subscriptions.forEach((subscription) => subscription.unsubscribe());
+  }
+
   public rewindVideo(value: number): void {
     this.currentTime = this.videoPlayerService.getCurrentTime();
     this.currentTime += value;
     this.videoPlayerService.setCurrentTime(this.currentTime);
     this.videoPlayerService.setCurrentTimeRewindFlag(true);
   }
```
Comment (Vladissqq) [inline src/app/pages/video-player/video-controllers/video-controllers.component.ts:35]: where is toggleFullscreen now?

File: src/app/pages/video-player/video-controllers/video-settings/video-settings.component.html [hunk 6]
```diff
@@ -19,25 +19,29 @@
       [(isPopupOpened)]="isPopupOpened"
     ></quality-settings>
   </div>
-  <button class="video-settings__btn" (click)="toggleContent()"></button>
+  <div class="video-settings__btn-container" (click)="toggleContent()">
```
Comment (Vladissqq) [inline src/app/pages/video-player/video-controllers/video-settings/video-settings.component.html:22]: same problem, read more about BEM

File: src/app/pages/video-player/video-player.component.ts [hunk 7]
```diff
@@ -52,27 +54,33 @@ export class VideoPlayerComponent implements OnInit, AfterViewInit {
...
-    this.videoPlayerService.isCurrentTimeRewound$.subscribe(
-      (isCurrentTimeRewound: boolean): void => {
-        this.isCurrentTimeRewound = isCurrentTimeRewound;
-      },
+    this.subscriptions.push(
+      this.videoPlayerService.isCurrentTimeRewound$.subscribe(
+        (isCurrentTimeRewound: boolean): void => {
+          this.isCurrentTimeRewound = isCurrentTimeRewound;
+        },
+      ),
```
Comment (Vladissqq) [inline src/app/pages/video-player/video-player.component.ts:83]: i think all this code u can split on independent functions, and declare them like methods of class

File: src/app/pages/video-player/video-player.component.ts [hunk 8]
```diff
@@ -107,10 +119,10 @@ export class VideoPlayerComponent implements OnInit, AfterViewInit {
...
   public toggleContent(): void {
     this.isMouseMove = true;
-    clearTimeout(this.timeout);
-    this.timeout = setTimeout(() => {
-      this.isMouseMove = !this.isMouseMove;
-    }, 3000);
+    // clearTimeout(this.timeout);
+    // this.timeout = setTimeout(() => {
+    //   this.isMouseMove = !this.isMouseMove;
+    // }, 3000);
```
Comment (Vladissqq) [inline src/app/pages/video-player/video-player.component.ts:125]: uncomment or delete)

**Review by Vladissqq — CHANGES_REQUESTED** [review]

File: src/app/pages/video-player/video-controllers/video-controllers.component.ts (hunk 5 above)
Comment (IvanSkadorva) [reply src/app/pages/video-player/video-controllers/video-controllers.component.ts:35]: In the separate component

File: src/app/pages/video-player/video-controllers/video-settings/video-settings.component.html (hunk 6 above)
Comment (IvanSkadorva) [reply src/app/pages/video-player/video-controllers/video-settings/video-settings.component.html:22]: Okay!

File: src/app/pages/video-player/video-controllers/video-controllers.component.scss (hunk 4 above)
Comment (IvanSkadorva) [reply src/app/pages/video-player/video-controllers/video-controllers.component.scss:8]: Hmm, &__btns-bar
Is it good?

File: src/app/pages/video-player/video-controllers/fullscreen-btn/fullscreen-btn.component.html (hunk 2 above)
Comment (IvanSkadorva) [reply src/app/pages/video-player/video-controllers/fullscreen-btn/fullscreen-btn.component.html:1]: Thank you, I'll fix it)

File: src/app/components/choose-season/choose-season.component.html [hunk 9]
```diff
@@ -20,6 +20,7 @@
   </choose-episode>
 </section>
 
-<div class="open-seasons__btn-container" (click)="onOpenSeasons()">
-  <button class="open-seasons__btn"></button>
-</div>
+<setting-btn
+  [imageSrc]="seasonImageSrc"
+  (click)="onOpenSeasons()"
+></setting-btn>
```
Comment (Vladissqq) [inline src/app/components/choose-season/choose-season.component.html:26]: i mean here btn-container it is block and btn it is element, className for this btn - btn-container__btn. but for div i think better use setting-btn-container or something like that

**Review by Vladissqq — DISMISSED** [review]

**Review by Vladissqq — APPROVED** [review]

**Pull request merged**